apps/dsb-message-broker/src/channel/channel.service.ts

```diff
@@ -22,42 +21,47 @@ export class ChannelService implements OnModuleInit {
         });
     }
 
-    public async create(
+    public async createChannel(
         channelData: CreateChannelDto & { createdBy: string; createdDateTime: string }
     ): Promise<string> {
-        const { org, app, channel } = extractFqcn(channelData.fqcn);
-        const organizations = this.configService.get('ORGANIZATIONS');
-        const fqcnIsValid = organizations.some((_org: any) => {
-            if (_org.name !== org) return false;
-            return _org.apps.some((_app: any) => {
-                if (_app.name !== app) return false;
-
-                return new RegExp(_app.channels).test(channel);
-            });
-        });
-        if (!fqcnIsValid) throw new Error('fqcn does not match the defined pattern.');
+        if (!channelData.admins) channelData.admins = [channelData.createdBy];
 
         return this.transport.createChannel(channelData);
     }
 
-    public async getAvailableChannels(userDID: string, userVR: string[]): Promise<Channel[]> {
+    public async updateChannel(
+        channelData: CreateChannelDto & { modifiedBy: string; modifiedDateTime: string }
+    ): Promise<string> {
+        const _channel = this.transport.getChannel(channelData.fqcn);
+        if (!_channel) throw new ChannelNotFoundError(channelData.fqcn);
+        const canModify = _channel.admins.some((_admin) => _admin === channelData.modifiedBy);
```

Comment:
couldn't a user get around this by setting the UpdateChannelDTO.admins to `[<MY_DID>]`?

Reply:
no, because it checks the did of requestor with admins in the addressbook and not the request body